src/main/java/com/laytonsmith/core/functions/Threading.java

```diff
@@ -329,4 +336,137 @@ public Version since() {
 
 	}
 
+	@api
+	@noboilerplate
+	@seealso({x_new_thread.class})
+	public static class _synchronized extends AbstractFunction {
+		private static final HashMap<Object, Integer> syncObjectMap = new HashMap<Object, Integer>();
```

Comment:
Use Map on the left side. Also, a TreeMap might be a better choice here, though actually, it probably won't matter, since the usual case will be 0 or 1 objects in here. Anyways, I would need to research it more, but you might need to use the _volatile_ keyword here. Or maybe that just applied to primitives. Anyways, go look that up.